src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -100,21 +100,21 @@ public static partial class JsonSerializer
                         }
                         else if (HandleValue(tokenType, options, ref reader, ref state))
                         {
-                            return;
+                            break;
                         }
                     }
                     else if (tokenType == JsonTokenType.EndArray)
                     {
                         if (HandleEndArray(options, ref state, ref reader))
                         {
-                            return;
+                            break;
                         }
                     }
                     else if (tokenType == JsonTokenType.Null)
                     {
                         if (HandleNull(ref reader, ref state, options))
                         {
-                            return;
+                            break;
```

Comment:
Effectively, instead of all these being `return`/`break`, can all these be `continue`, and then we can remove the call to `ConsumeTrailingWhitespaceAndComments`?

Basically, we need to keep calling `Read()` if we keep hitting a comment in the `AllowComments` mode and then, (regardless of the mode), we need to call `Read()` just one more time to move past the trailing whitespace/comments.

Reply:
Your suggestion works. The remaining nit is to handle the case when state.PropertyPath would throw an NPE when there is invalid JSON beyond the end.